components/search/backend/src/routes/kotlin/routes/GetRunsWithPackage.kt

```diff
@@ -89,40 +87,24 @@ internal fun Route.getRunsWithPackage(searchService: SearchService) =
                 }
             }
         }
-    }) {
+    }, requireScopedReadPermission()) {
         val identifierParam = call.requireParameter("identifier")
-        val organizationIdParam = call.request.queryParameters["organizationId"]?.toLongOrNull()
-        val productIdParam = call.request.queryParameters["productId"]?.toLongOrNull()
-        val repositoryIdParam = call.request.queryParameters["repositoryId"]?.toLongOrNull()
+        val scopeResult = call.parseScope()
```

Comment:
Related to my previous comment. I would propose to handle invalid scopes in the checker returned by `requireScopedReadPermission()` and use some exception mapping logic in [StatusPages](https://github.com/eclipse-apoapsis/ort-server/blob/main/core/src/main/kotlin/plugins/StatusPages.kt) to yield a 400 BAD RESPONSE code.